apps/skde/src/charts/MedfieldTable/index.tsx

```diff
@@ -0,0 +1,220 @@
+
+export type MedfieldTableProps = {
+  unitNames: string[];
+  treatmentYear: number;
+  context: string;
+  type: string;
+  width: number;
+};
+
+export type IndicatorLevels = {
+  ind_id: string;
+  year: number;
+  level: "H" | "M" | "L";
+};
+
+const createSymbols = (green: number, yellow: number, red: number) => {
+  let symbols = [];
+
+  for (let i = 0; i < green; i++) {
+    symbols.push(<FaCircle style={{ color: "#3baa34", fontSize: "1.2rem" }} />);
+  }
+
+  for (let i = 0; i < yellow; i++) {
+    symbols.push(<FaAdjust style={{ color: "#fd9c00", fontSize: "1.2rem" }} />);
+  }
+
+  for (let i = 0; i < red; i++) {
+    symbols.push(
+      <FaRegCircle style={{ color: "#e30713", fontSize: "1.2rem" }} />,
+    );
+  }
+
+  return symbols;
+};
+
+type RowData = {
+  name: string;
+  green: number;
+  yellow: number;
+  red: number;
+  registers: {
+    name: string;
+    green: number;
+    yellow: number;
+    red: number;
+  }[];
+};
+
+export const createMedfieldTableData = (data: Indicator[]) => {
+  // Set indicator colour from value and colour limits
+  const levels = data.map((row) => {
+    const indicatorLevel = level(row);
+    return {
+      ind_id: row.ind_id,
+      registry_id: row.registry_id,
+      registry_full_name: row.registry_full_name,
+      medfield_id: row.medfield_id,
+      medfield_full_name: row.medfield_full_name,
+      level: indicatorLevel,
+    };
+  });
+
+  // Group by medfield and registry and initialise counts
+  const rowData: RowData[] = levels.reduce((result, value) => {
+    if (!result[value.medfield_id]) {
+      result[value.medfield_id] = {
+        name: value.medfield_full_name,
+        green: 0,
+        yellow: 0,
+        red: 0,
+        registers: [],
+      };
+    }
+
+    result[value.medfield_id].registers[value.registry_id] = {
+      name: value.registry_full_name,
+      green: 0,
+      yellow: 0,
+      red: 0,
+    };
+
+    return result;
+  }, []);
+
+  // Count levels
+  for (let i = 0; i < levels.length; i++) {
+    const registry_id = levels[i].registry_id;
+    const medfield_id = levels[i].medfield_id;
+    const level = levels[i].level;
+
+    if (level == "H") {
+      rowData[medfield_id].green += 1;
+      rowData[medfield_id].registers[registry_id].green += 1;
+    }
+
+    if (level == "M") {
+      rowData[medfield_id].yellow += 1;
+      rowData[medfield_id].registers[registry_id].yellow += 1;
+    }
+
+    if (level == "L") {
+      rowData[medfield_id].red += 1;
+      rowData[medfield_id].registers[registry_id].red += 1;
+    }
+  }
+
+  return rowData;
+};
+
+const Row = (props: { row: RowData }) => {
+  const { row } = props;
+  const [open, setOpen] = React.useState(false);
+
+  return (
+    <React.Fragment>
+      <TableRow sx={{ "& > *": { borderBottom: "unset" } }}>
+        <TableCell>
+          <IconButton
+            aria-label="expand row"
+            size="small"
+            onClick={() => setOpen(!open)}
```

Comment:
Flytte `onClick={() => setOpen(!open)}` til `<TableRow>` slik at man ikke trenger å treffe akkurat på symbolet for å treffe?